core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceSettingService.scala

```diff
@@ -270,4 +276,17 @@ class WorkspaceSettingService(protected val ctx: RawlsRequestContext,
         }
       _ <- iamPolicyAction.compile.drain.unsafeToFuture()
     } yield ()
+
+  /**
+   * Call to handle entity attributes migration when compact data tables setting enabled.
+   */
+  private def applyCompactDataTablesSetting(workspace: Workspace): Future[Unit] =
+    entityService
+      .quicksilverMigration(workspaceName = WorkspaceName(workspace.namespace, workspace.name), updateSettings = false)
+      .map(_ => ())
+      .recover { case e: Exception =>
+        throw new RawlsExceptionWithErrorReport(
+          ErrorReport(StatusCodes.InternalServerError, s"Quicksilver migration failed: ${e.getMessage}")
+        )
+      }
```

Comment:
I think this should be asynchronous. Else, for big workspaces, the request to change workspace settings could take multiple minutes